File: sdk/dotnet/src/xjustiz.core-dotnet/Models/Entities/Core/DokumentCore.cs
namespace xjustiz.core_dotnet.Models.Entities.Core;

using System.Xml.Serialization;
using xjustiz.core_dotnet.Models.Helpers;
using xjustiz.core_dotnet.Util.Versioning;

/// <summary>
/// Repräsentiert ein Dokument.<br/>
/// <u><b>Document:</b></u> Represents a document.
/// </summary>
[XJustizCoreAvailability(XJustizCoreVersion.V0_2_0)]
public class DokumentCore : Dokument
{
    /// <summary>
    /// Fachspezifische Daten eines Dokuments.<br/>
    /// <u><b>Subject-specific data:</b></u> Specialized data of a document.
    /// </summary>
    [XmlElement("xjustiz.fachspezifischeDaten", Namespace = SchemeConstants.XJustiz_Tns, Order = 2)]
    [XJustizCoreAvailability(XJustizCoreVersion.V0_2_0)]
    public new XjustizFachspezifischeDatenCore FachspezifischeDaten { get; set; } = new();
}

File: sdk/java/src/main/java/de/xjustiz/core/models/DokumentCore.java
package de.xjustiz.core.models;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonInclude;
import jakarta.xml.bind.annotation.XmlAccessType;
import jakarta.xml.bind.annotation.XmlAccessorType;
import jakarta.xml.bind.annotation.XmlType;

/**
 * Repräsentiert ein Dokument.
 * <p>
 * <u><b>Document:</b></u> Represents a document.
 */
@XmlType
@XmlAccessorType(XmlAccessType.FIELD)
@JsonIgnoreProperties(ignoreUnknown = true)
@JsonInclude(JsonInclude.Include.NON_EMPTY)
public class DokumentCore extends Dokument {

    /**
     * Default constructor.
     */
    public DokumentCore() {}

}